src/pages/settings/Wallet/ActivatePhysicalCardPage.tsx

```diff
@@ -66,19 +65,18 @@ function ActivatePhysicalCardPage({
      * If state of the card is CONST.EXPENSIFY_CARD.STATE.OPEN, navigate to card details screen.
      */
     useEffect(() => {
-        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-        if (physicalCard?.isLoading || cardList?.[cardID]?.state !== CONST.EXPENSIFY_CARD.STATE.OPEN) {
+        if (physicalCard?.state !== CONST.EXPENSIFY_CARD.STATE.OPEN || physicalCard?.isLoading) {
             return;
         }
 
-        Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARD.getRoute(domain));
-    }, [cardID, cardList, domain, physicalCard?.isLoading]);
+        Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARD.getRoute(domain, cardId));
+    }, [cardId, cardList, domain, physicalCard?.isLoading, physicalCard?.state]);
 
     useEffect(
         () => () => {
-            CardSettings.clearCardListErrors(cardID);
+            CardSettings.clearCardListErrors(physicalCard?.cardID ?? 0);
```

Comment:
Hmm do we want to default to 0 here? It looks like it would merge 0 into the cardList

Reply:
Maybe not though? would 0: {} work?

Reply:
I could just do if and an early return. Dunno why I didn't think of that. Defaulting to 0 would probably cause bugs 😄

Reply:
Yeah let's do an early return instead